app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,23 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DiffUtil;
+
+public class BaseListDiffCallback<BaseListDiffCallback> extends DiffUtil.ItemCallback<BaseListDiffCallback>  implements ListDiffInterface<BaseListDiffCallback>{
```

Comment:
```suggestion
public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T>
```
Так будет правильнее, сам этот класс не должен наследовать ListDiffInterface